tests/qnodes/test_qnode_base.py

```diff
@@ -781,6 +781,39 @@ def circuit(w, x=None, **kwargs):
         c = node(1.0, x=np.pi, y=10)
         assert c == pytest.approx(-1.0, abs=tol)
 
+    def test_mutable_node_variable_number_of_args(self, qubit_device_1_wire):
+        """Tests a mutable circuit that expects a variable number of
+        positional arguments based on the values of its auxiliary arguments.
+        """
+        def circuit(x, n=1):
+            for k in range(n):
+                qml.RX(x[k], wires=0)
+            return qml.expval(qml.PauliZ(wires=0))
```

Comment:
This is another use-case where I would expect a user to simply do:
```python
def circuit(x):
    for p in x: # or for k in range(len(x)):
        qml.RX(p, wires=0)
     return qml.expval(qml.PauliZ(wires=0))
````

@mariaschuld I would like to get your thoughts on this, but this is a UI regression from v0.7 that I wouldn't like to lose.

Reply:
Technically `for p in x:` is ok as long as the qfunc is *always* called with an `x` that has the same number of elements, but it's not good style.

I seem to remember we discussed this issue when we were designing the qfunc parameter system, but I can't find an ADR that specifies what the qfuncs must look like.

Reply:
_We_ might know that this is no good style. But it's the most straightforward way of implementing this behaviour (I myself did it like this not knowing it was bad) and therefore the most natural way of doing it seen from the user side. And we should thus support it.

Reply:
I'm also in favour of allowing it, or documenting it very much clearly if we decide to disallow it

Reply:
If we allow flow control that depends on the positional parameter x it means we cannot know when the qfunc must be called again to reconstruct the circuit graph, and thus we would have to call it again at every evaluation to be safe. See  https://github.com/XanaduAI/pennylane-adrs/pull/2   :)

Reply:
See my comment in the `supermutable` thread.